src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerValidator.cs

```diff
@@ -664,6 +666,57 @@ public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
         }
     }
 
+    public class ComputedColumnNeedToBePersistedAndNotNullInPrimaryKeyRule : ITableDesignerValidationRule
+    {
+        public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
+        {
+            var table = designer.TableViewModel;
+            var errors = new List<TableDesignerIssue>();
+            if (table.PrimaryKey != null)
+            {
+                for (int i = 0; i < table.PrimaryKey.Columns.Count; i++)
+                {
+                    var columnSpec = table.PrimaryKey.Columns[i];
+                    var col = table.Columns.Items.First(item => item.Name == columnSpec.Column);
+                    if (col != null && col.IsComputed && !(col.IsComputedPersisted == true && col.IsComputedPersistedNullable == false))
+                    {
+                        errors.Add(new TableDesignerIssue()
+                        {
+                            Description = SR.ComputedColumnNeedToBePersistedAndNotNullInPrimaryKeyRuleDescription(col.Name),
+                            PropertyPath = new object[] { TablePropertyNames.PrimaryKeyColumns, i, IndexColumnSpecificationPropertyNames.Column }
+                        });
+                    }
+                }
+            }
+            return errors;
+        }
+    }
+
+    public class ComputedColumnNeedToBePersistedInForeignKeyRule : ITableDesignerValidationRule
+    {
+        public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
+        {
+            var table = designer.TableViewModel;
+            var errors = new List<TableDesignerIssue>();
+            for (int i = 0; i < table.ForeignKeys.Items.Count; i++)
+            {
+                for (int j = 0; j < table.ForeignKeys.Items[i].Columns.Count; j++)
+                {
+                    var col = table.Columns.Items.First(item => item.Name == table.ForeignKeys.Items[i].Columns[j]);
+                    if (col != null && col.IsComputed && col.IsComputedPersisted != true)
+                    {
+                        errors.Add(new TableDesignerIssue()
+                        {
+                            Description = SR.ComputedColumnNeedToBePersistedInForeignKeyRuleDescription(col.Name, table.ForeignKeys.Items[i].SystemName),
+                            PropertyPath = new object[] { TablePropertyNames.ForeignKeys, i, ForeignKeyPropertyNames.ColumnMapping, j, ForeignKeyColumnMappingPropertyNames.Column }
```

Comment:
this PropertyPath is set to the path of column specification in the FK, I'm not sure if this is better or should I point the user to the `IsPersisited` setting of the column directly.

Reply:
either one is fine, I don't have a preference.